### PR TITLE
Game loop cv

### PR DESCRIPTION
**Pull request by nashirj**
This PR sets up the basic code infrastructure for computer vision. We still need to be able to detect board corners (#100), classify piece type for promotion (#101), and piece occupancy for general board state detection (#102)

### CODE REVIEW

**Comment by nashirj** [comment]
Just realized there are a few things in boardinterface that need to be cleaned up still. I'll message when I'm ready for you to take a look

File: firmware/raspi/boardinterface.py [hunk 1]
```diff
@@ -73,16 +73,6 @@ def is_capture(self, uci_move):
...
-    # def get_chess_coords_from_square(square):
-    #     '''Converts chess square to a BoardCell.
-
-    #     Example: a1 <=> [0, 0], h8 <=> [7, 7], regardless of whether human plays white or black
-    #     pieces.
-    #     '''
-    #     # Nums correspond to row (rank), letters correspond to col (files)
-    #     return util.BoardCell(ord(square[1]) - ord('1'), ord(square[0]) - ord('a'))
-
     def get_board_coords_from_square(self, square):
```
Comment (ColonelToad) [inline firmware/raspi/boardinterface.py:76]: Why did you decide get_chess_coords_from_square should be moved at the end of the uti.py file?

Comment (nashirj) [reply firmware/raspi/boardinterface.py:76]: I see now that I didn't need to. My thinking was that I didn't want to do `from boardinterface import Engine` in the `vision` file. Would you prefer I do that instead? I can change it back.

Comment (ColonelToad) [reply firmware/raspi/boardinterface.py:76]: No, I think that was a good change and was just wondering why.

Comment (nashirj) [reply firmware/raspi/boardinterface.py:76]: Sounds good!

**Review by JuddBE — APPROVED** [review]
Looks good to me!

**Comment by nashirj** [comment]
Thanks for the review!

**Pull request merged**